src/module/flows/overheat.ts

```diff
@@ -0,0 +1,340 @@
+
+export function registerOverheatSteps(flowSteps: Map<string, Step<any, any> | Flow<any>>) {
+  flowSteps.set("preOverheatRollChecks", preOverheatRollChecks);
+  flowSteps.set("rollOverheatTable", rollOverheatTable);
+  flowSteps.set("noStressRemaining", noStressRemaining);
+  flowSteps.set("checkOverheatMultipleOnes", checkOverheatMultipleOnes);
+  flowSteps.set("overheatInsertEngCheckButton", overheatInsertEngCheckButton);
+  flowSteps.set("overheatInsertCascadeRollButton", overheatInsertCascadeRollButton);
+  flowSteps.set("printOverheatCard", printOverheatCard);
+}
+
+/**
+ * OverheatFlow manages all the steps necessary for the initial overheat rolls and outcomes.
+ */
+export class OverheatFlow extends Flow<LancerFlowState.OverheatRollData> {
+  name = "OverheatFlow";
+  steps = [
+    "preOverheatRollChecks",
+    "rollOverheatTable",
+    "noStressRemaining",
+    "checkOverheatMultipleOnes",
+    "overheatInsertEngCheckButton",
+    "overheatInsertCascadeRollButton",
+    "printOverheatCard",
+  ];
+
+  constructor(uuid: UUIDRef | LancerActor, data?: Partial<LancerFlowState.OverheatRollData>) {
+    const initialData: LancerFlowState.OverheatRollData = {
+      type: "overheat",
+      title: data?.title ?? "",
+      roll_str: data?.roll_str ?? "",
+      desc: data?.desc ?? "",
+      val: data?.val ?? -1,
+      max: data?.max ?? -1,
+      remStress: data?.remStress ?? 4,
+    };
+
+    super(uuid, initialData);
+  }
+}
+
+/**
+ * Performs initial checks and verifications that should be made BEFORE rolling for overheat.
+ *
+ * @param actor   - Actor or ID of actor to overheat
+ * @param reroll_data - Data to use if rerolling. Setting this also supresses the dialog.
+ */
+export async function preOverheatRollChecks(state: FlowState<LancerFlowState.OverheatRollData>): Promise<boolean> {
+  if (!state.data) throw new TypeError(`Overheat roll flow data missing!`);
+  const actor = state.actor;
+  if (!actor.is_mech() && !actor.is_npc()) {
+    ui.notifications!.warn("Only Mechs and NPCs can take stress damage");
+    return false;
+  }
+
+  if (getAutomationOptions().structure && !state.data?.reroll_data) {
+    if (actor.system.heat.value <= actor.system.heat.max) {
+      ui.notifications!.info("Token is not at heat cap. No need to roll stress.");
+      return false;
+    }
+    const { openSlidingHud: open } = await import("../helpers/slidinghud");
+    try {
+      await open("stress", { stat: "stress", title: "Stress Damage", lancerActor: actor });
+    } catch (_e) {
+      // User hit cancel, abort the flow.
+      return false;
+    }
+  }
+
+  // After the hud is closed, update the actor's heat and stress.
+  if (!state.data?.reroll_data) {
+    let heat = actor.system.heat;
+    let stress = actor.system.stress;
+    if (heat.value > actor.system.heat.max && stress.value > 0) {
+      await actor.update({
+        "system.stress": stress.value - 1,
+        "system.heat": heat.value - actor.system.heat.max,
```

Comment:
There's a special case for single-structure NPCs that currently isn't supported, but I think could be better supported fairly easily. When they go over heat cap they don't lose structure, they instead become exposed until they clear their heat.

We should copy the core book's wording for that case and display it in the card. Do you think it makes sense to add that to `overheatTableTitles` and `overheatTableDescriptions`? Or should we treat it as a truly special case and have separate variables that we use?